### PR TITLE
SG simplification

### LINKED ISSUE #275: Reduce Complexity of Security Group Rules
### Improvement

It seems like there is an opportunity to reduce complexity in terms of number of tables required to fully define ec2 Security Groups. Currently there are 10 tables to fully define egress and ingress rules. These tables could be collapsed into 5 tables where we would just add a column to specify if it is an egress or ingress rule. The value is already returned from the `describe-security-group-rules` call in `IsEgress`

### Use Case

Reduce complexity when writing queries and just use `where` clauses rather than joins

### PR DESCRIPTION
**Pull request by bbernays**
Closes #275 

ToDo:
- [x] update integration tests
- [x] Write migrations

Output from acceptance testing:
More information about running the tests [here](../docs/contributing/e2e_tests.md)

```
INTEGRATION_TESTS=1 TF_VAR_PREFIX=cq-testing TF_APPLY_RESOURCES=0 TF_VAR_SUFFIX=integration go test -timeout 180m -v -run ^TestIntegrationEc2SecurityGroups  github.com/cloudquery/cq-provider-aws/resources/integration_tests
=== RUN   TestIntegrationEc2SecurityGroups
=== PAUSE TestIntegrationEc2SecurityGroups
=== CONT  TestIntegrationEc2SecurityGroups
2021/12/21 16:21:40 aws_ec2_security_groups verify fields
2021/12/21 16:21:40 aws_ec2_security_groups fetch resources
2021-12-21T16:21:42.672-0500 [INFO]  fetched successfully: Region=us-east-1 account_id=7049xxxxxxxx table=aws_ec2_security_groups count=24
--- PASS: TestIntegrationEc2SecurityGroups (2.32s)
PASS
ok      github.com/cloudquery/cq-provider-aws/resources/integration_tests       3.017s
```

### CODE REVIEW

File: resources/ec2_security_groups.go [hunk 1]
```diff
@@ -402,83 +235,74 @@ func resolveEc2securityGroupTags(ctx context.Context, meta schema.ClientMeta, re
...
-	res <- securityGroupIpPermission.Ipv6Ranges
+	res <- ipRanges
 	return nil
 }
-func fetchEc2SecurityGroupIpPermissionPrefixListIds(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
-	securityGroupIpPermission, ok := parent.Item.(types.IpPermission)
-	if !ok {
-		return fmt.Errorf("not ec2 security group ip permission")
+
+func getString(s *string) string {
```
Comment (disq) [inline resources/ec2_security_groups.go:258]: aws sdk already has a fn for that, `aws.String(*string) string`

Comment (bbernays) [reply resources/ec2_security_groups.go:258]: Thank you! it is `aws.ToString`

**Review by disq — APPROVED** [review]

**Review by yevgenypats — APPROVED** [review]

**Comment by yevgenypats** [comment]
LGTM. I suggest merging this after we move to the new testing framework

**Review by roneli — APPROVED** [review]
LGTM, would upgrade to latest changes and update the up/down sql to v0.9.1

**Comment by yevgenypats** [comment]
@bbernays can you please solve the conflicts and we will merge? now that we released all the policies. Sorry for the delay hopefully now with the semantic version pinning in policies we can push things faster :) .

**Review by roneli — APPROVED** [review]
LGTM

**Pull request merged**